src/applications/app-errors.js

```diff
@@ -14,15 +14,23 @@ export function handleAppError(err, app) {
 
 export function addErrorHandler(handler) {
   if (typeof handler !== "function") {
-    throw Error("a single-spa error handler must be a function");
+    throw Error(
+      __DEV__
+        ? devErrorMessage(28, "a single-spa error handler must be a function")
```

Comment:
I considered using a more complex approach that included a babel plugin and a managed set of error codes. However getting that babel plugin to be perfect was tricky. So I decided to try out just a manual process where when we add a new error we have to find and create a new error code. And we have to write the `__DEV__` ternary ourselves instead of the babel plugin.

Reply:
I don't mind doing it ourselves. 

I do wonder if we could make it simpler though -- let me know what you think about this:

```js
throw Error(sspaErrorMessage(28, __DEV___ && "error message")
```
and combine the error functions into a single function? 

```js
export function sspaErrorMessage(code, msg, ...args) {
  return `single-spa error #${code} ${msg ? msg : ""}. See https://single-spa.js.org/error?code=${code}${
    args.length ? `&arg=${args.join("&arg=")}` : ""
  }`;
}
```

don't know if get the same size benefits from this or not, but could potentially be a bit easier to work with? 

https://rollupjs.org/repl/?version=1.27.8&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwX19ERVZfXyUyMCUzRCUyMGZhbHNlJTVDbiU1Q25mdW5jdGlvbiUyMHRlc3RpbmcobnVtJTJDJTIwbWVzc2FnZSklMjAlN0IlNUNuJTVDdHJldHVybiUyMCU2MG51bSUzQSUyMCUyNCU3Qm51bSU3RCUyMCUyNCU3Qm1lc3NhZ2UlMjAlM0YlMjBtZXNzYWdlJTIwJTNBJTIwJTVDJTIyJTVDJTIyJTdEJTYwJTVDbiU3RCU1Q24lNUNudGVzdGluZyhudW0lMkMlMjBfX0RFVl9fJTIwJTI2JTI2JTIwJTVDJTIyaGVsbG8lNUMlMjIpJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

Reply:
👍 That's easier, good idea. I'll change it over.